File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/ApiKeyResponse.ts
import { ResponseModelBase } from '../../ResponseModelBase'
import type { ApiKeyEntityScopeResponse } from './ApiKeyEntityScopeResponse'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.ApiKeyResponse</c>. Plaintext <c>apiKey</c> when issued or loaded per server rules. */
export interface ApiKeyResponse extends ResponseModelBase {
  id: string
  apiKey: string
  createdAt: string
  description?: string | null
  expiresAt?: string | null
  isGlobalAdmin: boolean
  entityScopes?: ApiKeyEntityScopeResponse[] | null
}

File: src/MaksIT.CertsUI/Models/APIKeys/ApiKeyResponse.cs
using MaksIT.Core.Abstractions.Webapi;
using MaksIT.CertsUI.Engine;


namespace MaksIT.CertsUI.Models.APIKeys;

public class ApiKeyResponse : ResponseModelBase {
  public Guid Id { get; set; }
  public required string ApiKey { get; set; }
  public DateTime CreatedAt { get; set; }
  public string? Description { get; set; } // Optional description for the API key
  public DateTime? ExpiresAt { get; set; }
  public bool IsGlobalAdmin { get; set; }
  public List<ApiKeyEntityScopeRsponse>? EntityScopes { get; set; }
}
